src/VisualStudio/Core/Next/Remote/ServiceHubRemoteHostClient.cs

```diff
@@ -136,6 +160,37 @@ private void OnRpcDisconnected(object sender, JsonRpcDisconnectedEventArgs e)
             Disconnected();
         }
 
+        private static async Task<TResult> RetryRemoteCallAsync<TInput, TException, TResult>(
+            Func<TInput, CancellationToken, Task<TResult>> funcAsync,
+            TInput input,
+            TimeSpan timeout,
+            CancellationToken cancellationToken) where TException : Exception
```

Comment:
i don't understand the complexity of this signature.  Why not just be:

```c#
Task<TResult> RetryRemoteCallAsync<TException, TResult>(
 +            Func<Task<TResult>> funcAsync,
 +            TimeSpan timeout,
 +            CancellationToken cancellationToken)
```
?

All the input passing can happen at the callsite like:

```c#
RetryRemoteCallAsync(() => CreateWorkerAsync(workspace, primary, timeout), cancellationToken);
```

Reply:
if we are okay with capturing sure.